### PR TITLE
Add get tags

### PR DESCRIPTION
**Pull request by jinliu9508**
# Description
## One Line Summary
Add getTags that returns a map of key-value tags

### Motivation
Allow user to get access of tags

# Affected code checklist
   - [ ] Notifications
      - [ ] Display
      - [ ] Open
      - [ ] Push Processing
      - [ ] Confirm Deliveries
   - [ ] Outcomes
   - [ ] Sessions
   - [ ] In-App Messaging
   - [ ] REST API requests
   - [ ] Public API changes

# Checklist
## Overview
   - [x] I have filled out all **REQUIRED** sections above
   - [x] PR does one thing
     - If it is hard to explain how any codes changes are related to each other then it most likely needs to be more than one PR
   - [ ] Any Public API changes are explained in the PR details and conform to existing APIs

## Testing
   - [x] I have included test coverage for these changes, or explained why they are not needed
   - [x] All automated tests pass, or I explained why that is not possible
   - [x] I have personally tested this on my device, or explained why that is not possible

## Final pass
   - [x] Code is as readable as possible.
      - Simplify with less code, followed by splitting up code into well named functions and variables, followed by adding comments to the code.
   - [x] I have reviewed this PR myself, ensuring it meets each checklist item
      - WIP (Work In Progress) is ok, but explain what is still in progress and what you would like feedback on. Start the PR title with "WIP" to indicate this.

- - -
This change is [Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671)

### CODE REVIEW

**Review by shepherd-l — CHANGES_REQUESTED** [review]
Will the iOS implementation of getTags be included in this PR?
 
In the Unity SDK, we have a [CHANGELOG](https://github.com/OneSignal/OneSignal-Unity-SDK/blob/user-model/main/OneSignalExample/Assets/OneSignal/CHANGELOG.md) - could you update it as well

I recommend manually testing the getTags methods on Android and iOS with the example app
You can also add a getTags button to the example app to make testing easier in the future

Nit: could you also update the Migration Guide to include the new method

> *[Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-:-NkvqmRU3O1M-cgaAqtG:b-upx8gy)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @jinliu9508)

___
*[`OneSignalExample/Assets/StreamingAssets/SwiftPlugin.framework.meta` line 0 at r1](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-NkvqYyw43vRg37PU8VN:-NkvqYyw43vRg37PU8VO:b3iuvey) ([raw file](https://github.com/OneSignal/OneSignal-Unity-SDK/blob/6a4d29b012421bfbf731297355ef0cac7d33428a/OneSignalExample/Assets/StreamingAssets/SwiftPlugin.framework.meta#L0)):*
What does deleting this file do?

File: OneSignalExample/Assets/OneSignal/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 - Included meta files in OneSignalConfig.androidlib to prevent asset from being ignored
 - Package download url in the "Sync example code bundle package" step from the OneSignal SDK Setup
+- Add public get tags method
```
Comment (jkasten2) [inline OneSignalExample/Assets/OneSignal/CHANGELOG.md:22]: This should be under the "Changed" section instead of "Fixed".

**Review by jkasten2 — CHANGES_REQUESTED** [review]
Let's revert the changes to `OneSignalExampleScene.unity` in this PR, as it doesn't look related.

**Comment by jinliu9508** [comment]
> Let's revert the changes to `OneSignalExampleScene.unity` in this PR, as it doesn't look related.

I added a testing button "Get Tags" under the button "Remove Tag" in the example scene so tester can see what getTags() returns. This is a generated file so there are code changes that don't look relevant.

**Review by shepherd-l — APPROVED** [review]
Looks good! I have some minor nits
I tested GetTags on an emulated Pixel 4 with Android 12 and physical iPhone 12 with iOS 15.5 and the correct values were returned

> Reviewed 3 of 4 files at r1, 4 of 11 files at r3, 4 of 8 files at r4.
> *[Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-:-Nnu_f2A9D_SC-FeUkUu:b-yaf3g1)* status: 11 of 18 files reviewed, 7 unresolved discussions (waiting on @jinliu9508, @jkasten2, and @nan-li)

___
*[`com.onesignal.unity.android/Runtime/AndroidUserManager.cs` line 57 at r4](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-NnuMFoTEE-4Y7Gbczm0:-NnuMFoTEE-4Y7Gbczm1:bb7q3l2) ([raw file](https://github.com/OneSignal/OneSignal-Unity-SDK/blob/bd2c14bf29c780d62786cb5c7c4c5cc0c5140e29/com.onesignal.unity.android/Runtime/AndroidUserManager.cs#L57)):*
> ```Smalltalk
>             AndroidJavaObject obj = _user.Call<AndroidJavaObject>("getTags");
>             Dictionary<string, string> dictionary = obj.MapToDictionary();
>             UnityEngine.Debug.Log(dictionary.Count);
> ```

I think you forgot to remove a debug log, could you remove it?
Since the "dictionary" variable is only used for the log, you could also remove it or decide to return it instead
___
*[`OneSignalExample/Assets/OneSignal/CHANGELOG.md` line 19 at r4](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-NnuZBMI7RhipqhqMsi2:-NnuZBMI7RhipqhqMsi3:b5caiv5) ([raw file](https://github.com/OneSignal/OneSignal-Unity-SDK/blob/bd2c14bf29c780d62786cb5c7c4c5cc0c5140e29/OneSignalExample/Assets/OneSignal/CHANGELOG.md#L19)):*
> ```Markdown
> 
> ## [5.0.5]
> ### Changed
> ```

I think we should also add that we updated the OneSignal Android and iOS SDK to highlight native changes

It should look something like this. You can also see the same comment in the Changelog history
- Updated included Android SDK to \[5.0.5\](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/5.0.5)
- Updated included iOS SDK to \[5.0.5\](https://github.com/OneSignal/OneSignal-iOS-SDK/releases/tag/5.0.5)
___
*[`OneSignalExample/Assets/OneSignal/MIGRATION_GUIDE_v3_to_v5.md` line 177 at r4](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-NnuSE77CZKGvEZ_5C4v:-NnuSE77CZKGvEZ_5C4w:biecnix) ([raw file](https://github.com/OneSignal/OneSignal-Unity-SDK/blob/bd2c14bf29c780d62786cb5c7c4c5cc0c5140e29/OneSignalExample/Assets/OneSignal/MIGRATION_GUIDE_v3_to_v5.md#L177)):*
> ```Markdown
> | `void AddSms(string sms)`                             | *Add a new SMS subscription to the current user.*                                                                                                                                                                                        |
> | `void RemoveSms(string sms)`                          | *Remove an SMS subscription from the current user.*                                                                                                                                                                                      |
> | `Dictionary<string, string> GetTags()`                | *Return a copy of all tags for the current user. Tags are key:value pairs used as building blocks for targeting specific users and/or personalizing messages.*         |
> ```

I think we should also highlight that get tags returns the local tags on the user
___
*[`OneSignalExample/Assets/OneSignal/Example/OneSignalExampleBehaviour.cs` line 380 at r4](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-NnuRcFj6TvwQbsRfy44:-NnuRcFk5B8t0IipYF4F:b-exna63) ([raw file](https://github.com/OneSignal/OneSignal-Unity-SDK/blob/bd2c14bf29c780d62786cb5c7c4c5cc0c5140e29/OneSignalExample/Assets/OneSignal/Example/OneSignalExampleBehaviour.cs#L380)):*
> ```Smalltalk
>         Dictionary<string, string> dict = OneSignal.User.GetTags();
>         string dictionaryString = "{";
>         foreach(KeyValuePair < string, string > keyValues in dict) {  
> ```

Nit: could you remove the extra spaces next to the "<" and ">" to look like <string, string>
___
*[`OneSignalExample/Assets/OneSignal/Example/OneSignalExampleScene.unity` line 11343 at r4](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-NnuQ9DA9VzL9oQF-MJC:-NnuQ9DA9VzL9oQF-MJD:blvhfxp) ([raw file](https://github.com/OneSignal/OneSignal-Unity-SDK/blob/bd2c14bf29c780d62786cb5c7c4c5cc0c5140e29/OneSignalExample/Assets/OneSignal/Example/OneSignalExampleScene.unity#L11343)):*
> ```
>     m_VerticalOverflow: 0
>     m_LineSpacing: 1
>   m_Text: Get Tag
> ```

Nit but could you change the button to say "Get Tags" to match the GetTags method name
___
*[`OneSignalExample/Assets/StreamingAssets/SwiftPlugin.framework.meta` line  at r1](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-NkvqYyw43vRg37PU8VN:-NnuPje51hzbPmqs2k4K:b-2mi2ir) ([raw file](https://github.com/OneSignal/OneSignal-Unity-SDK/blob/6a4d29b012421bfbf731297355ef0cac7d33428a/OneSignalExample/Assets/StreamingAssets/SwiftPlugin.framework.meta#L)):*
<details><summary><i>Previously, shepherd-l (Shepherd) wrote…</i></summary><blockquote>

What does deleting this file do?
</blockquote></details>

Why was this file deleted? I don't think we need to delete this file

**Review by jinliu9508 — COMMENTED** [review]
I have committed a change addressing all of the concerns, please take a look again!

> Reviewed 3 of 4 files at r1, 7 of 11 files at r3, 8 of 8 files at r4.
> *[Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-:-NnuhA4eCXClFXFUJpaU:b-6c8vlr)* status: 11 of 18 files reviewed, 7 unresolved discussions (waiting on @jkasten2, @nan-li, and @shepherd-l)

**Review by jinliu9508 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671)* status: 11 of 18 files reviewed, 5 unresolved discussions (waiting on @jkasten2, @nan-li, and @shepherd-l)

___
*[`com.onesignal.unity.android/Runtime/AndroidUserManager.cs` line 57 at r4](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-NnuMFoTEE-4Y7Gbczm0:-NnuhNCA1nzhWJKp0weP:b-896fix) ([raw file](https://github.com/OneSignal/OneSignal-Unity-SDK/blob/bd2c14bf29c780d62786cb5c7c4c5cc0c5140e29/com.onesignal.unity.android/Runtime/AndroidUserManager.cs#L57)):*
<details><summary><i>Previously, shepherd-l (Shepherd) wrote…</i></summary><blockquote>

I think you forgot to remove a debug log, could you remove it?
Since the "dictionary" variable is only used for the log, you could also remove it or decide to return it instead
</blockquote></details>

Done.
___
*[`OneSignalExample/Assets/OneSignal/CHANGELOG.md` line 22 at r3](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#gh-1446750740:-NnuhTAM7ZsXUTmu-XWD:b-896fix) ([raw file](https://github.com/OneSignal/OneSignal-Unity-SDK/blob/8361d642c9046a510776ddf9087d2f90069ff8c0/OneSignalExample/Assets/OneSignal/CHANGELOG.md#L22)):*
<details><summary><i>Previously, jkasten2 (Josh Kasten) wrote…</i></summary><blockquote>

This should be under the "Changed" section instead of "Fixed".
</blockquote></details>

Done.
___
*[`OneSignalExample/Assets/OneSignal/CHANGELOG.md` line 19 at r4](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-NnuZBMI7RhipqhqMsi2:-NnuhRRACmfoui-q1agL:b-896fix) ([raw file](https://github.com/OneSignal/OneSignal-Unity-SDK/blob/bd2c14bf29c780d62786cb5c7c4c5cc0c5140e29/OneSignalExample/Assets/OneSignal/CHANGELOG.md#L19)):*
<details><summary><i>Previously, shepherd-l (Shepherd) wrote…</i></summary><blockquote>

I think we should also add that we updated the OneSignal Android and iOS SDK to highlight native changes

It should look something like this. You can also see the same comment in the Changelog history
- Updated included Android SDK to \[5.0.5\](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/5.0.5)
- Updated included iOS SDK to \[5.0.5\](https://github.com/OneSignal/OneSignal-iOS-SDK/releases/tag/5.0.5)
</blockquote></details>

Done.
___
*[`OneSignalExample/Assets/OneSignal/MIGRATION_GUIDE_v3_to_v5.md` line 177 at r4](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-NnuSE77CZKGvEZ_5C4v:-NnuhUH82EwBAEv3R9_o:b-896fix) ([raw file](https://github.com/OneSignal/OneSignal-Unity-SDK/blob/bd2c14bf29c780d62786cb5c7c4c5cc0c5140e29/OneSignalExample/Assets/OneSignal/MIGRATION_GUIDE_v3_to_v5.md#L177)):*
<details><summary><i>Previously, shepherd-l (Shepherd) wrote…</i></summary><blockquote>

I think we should also highlight that get tags returns the local tags on the user
</blockquote></details>

Done.
___
*[`OneSignalExample/Assets/StreamingAssets/SwiftPlugin.framework.meta` line  at r1](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-NkvqYyw43vRg37PU8VN:-Nnuhbtr5WGdJL2zlop6:b-896fix) ([raw file](https://github.com/OneSignal/OneSignal-Unity-SDK/blob/6a4d29b012421bfbf731297355ef0cac7d33428a/OneSignalExample/Assets/StreamingAssets/SwiftPlugin.framework.meta#L)):*
<details><summary><i>Previously, shepherd-l (Shepherd) wrote…</i></summary><blockquote>

Why was this file deleted? I don't think we need to delete this file
</blockquote></details>

Done.

**Review by shepherd-l — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/671#-:-Nnuw3i1Anwh5rBN24JE:bnfp4nl)* status: 11 of 18 files reviewed, 1 unresolved discussion (waiting on @jinliu9508, @jkasten2, and @nan-li)

**Pull request merged**